src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1165,9 +1180,9 @@ private static (BoundDagTemp input, BoundDagTemp lengthTemp, int index) GetCanon
             return (OriginalInput(input), lengthTemp, index);
         }
 
-        private static ImmutableArray<StateForCase> RemoveEvaluation(ImmutableArray<StateForCase> cases, BoundDagEvaluation e)
+        private static FrozenArrayBuilder<StateForCase> RemoveEvaluation(FrozenArrayBuilder<StateForCase> cases, BoundDagEvaluation e)
```

Comment:
nit: I much like the frozen-in and frozen-return in the signature, which makes it clearer that this must use a new instance, but feels like this method name could benefit from saying it returns a new instance. Maybe "GetInstanceMinusEvaluation"?

Reply:
I think it's fine as is. :)